server/controllers/users/userController.ts

```diff
@@ -0,0 +1,60 @@
+import { Response } from 'express';
+import Interviewee from '../../database/Models/Interviewee';
+import User from '../../database/Models/User';
+import { CustomError, RequestType } from '../../utils';
+
+const getAllReviews = async (req: RequestType, res: Response) => {
+  let { page, saved } = req.query;
+
+  if (!page || !saved) {
+    page = '1';
+    saved = 'false';
+  }
+  const id = req.userInfo?.id;
+  const user : object | null = await User.findById(id);
+  if (!user) {
+    throw new CustomError('User not found', 404);
+  }
+  if (!Number(page)) {
+    throw new CustomError('Invalid Page', 400);
+  }
+
+  if (saved !== 'true' && saved !== 'false') {
+    throw new CustomError('Invalid Input', 400);
+  }
+
+  // TODO: replace the userId string with the userId from the user object
+  const userInterviews = await Interviewee.find({ userId: '4' });
+
+  if (userInterviews.length === 0) {
+    return res.json({
+      message: 'No interviews found',
+    });
+  }
```

Comment:
Why do you search for the interviews first? you can immediately search for the reviews and if there are no interviews then it will return no reviews.